distribution/tools/plugin-cli/src/main/java/org/elasticsearch/plugins/cli/InstallPluginCommand.java

```diff
@@ -81,7 +95,55 @@ protected void execute(Terminal terminal, OptionSet options, Environment env) th
             .collect(Collectors.toList());
         final boolean isBatch = options.has(batchOption);
 
+        if (plugins.isEmpty()) {
+            throw new UserException(ExitCodes.USAGE, "at least one plugin ID is required");
+        }
+
         InstallPluginAction action = new InstallPluginAction(terminal, env, isBatch);
-        action.execute(plugins);
+        try {
+            action.execute(plugins);
+        } catch (InstallPluginException e) {
+            int exitCode;
+
+            switch (e.getProblem()) {
```

Comment:
Rather than manage this big switch statement here could we add an `exitCode` property to `InstallPluginProblem`, manage this mapping there, and then just return `e.getProblem().getExitCode()`?